extensions/functions_string.yaml

```diff
@@ -163,3 +164,23 @@ scalar_functions:
           - value: "fixedchar<L1>"
           - value: "varchar<L2>"
         return: "BOOLEAN"
+  -
+    name: lower
+    description: Transforms the string to lower case characters
```

Comment:
Let's add some additional definition here of how lower case is defined. I assume there is a utf definition?

Reply:
Arrow defers to utf8proc which is somewhat sparse on details.  However, you are correct, there is a standard UTF-8 way of lowercasing, though it [sometimes does the wrong thing semantically](https://github.com/JuliaLang/julia/issues/35615).